### PR TITLE
Updates: Android and workflows

### PR DESCRIPTION
**Pull request by PatrykMis**
I license this contribution under the terms set out in the MIT-0 license.

#### Android: build changes

Bumps:

* Android Gradle Plugin
* Desugar
* CURL
* Moshi

Enhancements:

* Add BuildTools version

Deprecations:

* move away from deprecated `package` attribute
* avoid deprecated functions and attributes

Fixes:

* Define ext.myNdkVersion and move it to proper place

#### Others

* Update CI
* Add Gradle Wrapper Validation Action
* Fix Dependabot configuration
* Reformat Android build scripts and JAVA code

### CODE REVIEW

**Comment by zstanecic** [comment]
great to see that you hopefully state the license.
Please state this in 
#556

**Comment by zstanecic** [comment]
@PatrykMis 
are you going also to implement direct boot support?

**Review by alex19EP — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,19 +24,20 @@ jobs:
...
           submodules: 'true'
 
-      - name: Set up JDK 1.11
+      - name: Set up JDK 18
         uses: actions/setup-java@v3
         with:
-          distribution: 'zulu'
-          java-version: '11'
+          distribution: 'temurin'
+          java-version: '18'
```
[the remark this answers is not in the document]
Comment (PatrykMis) [reply .github/workflows/ci.yml:31]: Java 18 at the moment is the latest stable version. It's obviously capable of running code designed for earlier versions while we can use latest runtime improvements. Note that source and target attributes are still 11, so it is mainly the runtime thing.
After all these changes (including gradle caching) Android build task times (also visible in Github actions) are decreased though.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,19 +24,20 @@ jobs:
...
+          cache: gradle
 
       - name: Install Scons
         run: |
           sudo apt-get -q -y update
           sudo apt-get -q -y install scons
 
       - name: Assemble app debug APK
-        run: ./gradlew assembleDebug --stacktrace
+        run: ./gradlew assembleDebug --stacktrace --no-daemon
```
[the remark this answers is not in the document]
Comment (PatrykMis) [reply .github/workflows/ci.yml:40]: [As stated in the Gradle's manual](https://docs.gradle.org/current/userguide/performance.html#enable_the_daemon_on_old_gradle_versions) daemon increases performance only when running more tasks than just one (like in our case). Each workflow we are running on a fresh machine though. It's also better for security to "fork [our] daemon".

[the remark this answers is not in the document]
Comment (alex19EP) [reply .github/workflows/ci.yml:40]: ok thanks

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -75,10 +76,10 @@ jobs:
         with:
           submodules: 'true'
 
-      - name: Set up Python 3.7
-        uses: actions/setup-python@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
         with:
-          python-version: 3.7
+          python-version: '3.10'
```
[the remark this answers is not in the document]
Comment (alex19EP) [reply .github/workflows/ci.yml:82]: pleas fix this

**Review by alex19EP — CHANGES_REQUESTED** [review]
one more thing and good to merge.

File: .github/workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (PatrykMis) [reply .github/workflows/ci.yml:82]: Python 3.7 is almost in eol (end of life) state. Is it really nessesary to use that old version?

**Comment by zstanecic** [comment]
Hi,

Yes, because of NVDA.

When NVDA switches to the new python, it will be reasonable to change this.

**Comment by Olga-Yakovleva** [comment]
This seems to be the reason for #658. Found by reverting Android related commits until release builds started working. I suspect the Android proguard rules replacement to the optimized set, but I am yet to test.

**Comment by Olga-Yakovleva** [comment]
Yes, already done in #659.

**Comment by zstanecic** [comment]
Are you going to re-enable alpha to test it, as it is relevant to releases and alpha?

**Comment by Olga-Yakovleva** [comment]
Yes, but not before I have tested it myself, uploaded to Google Play, 
got it reviewed by Google, and waited just in case, so that there is no 
chance for that old release to get to users and break speech on their 
devices.

**Pull request merged**